backend/app/auth/service.py

```
@@ -172,11 +203,17 @@ async def authenticate_with_google(self, id_token: str) -> Dict[str, Any]:
-            
+            try:
+                user = await db.users.find_one({"$or": [
+                    {"email": email},
+                    {"firebase_uid": firebase_uid}
+                ]})
+            except PyMongoError as e:
+                logger.error("Database error while checking user: %s", str(e))
+                raise HTTPException(
+                    status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                    detail="Internal server error"
+                )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)